cmake/precompiled_header.cmake

```diff
@@ -0,0 +1,26 @@
+# Configures sources on a target to use a precompiled header. This function takes a target and
```

Comment:
Please add license header and copyright notice.

Reply:
Added